Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Net.UnityServices.Infrastructure;
+using UnityEngine.SceneManagement;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Visual
+{
+    /// <summary>
+    /// responsible for driving all the functionality of the popup panel players see when connecting to the game
+    /// </summary>
+    public class PopupPanel : MonoBehaviour
+    {
+        [SerializeField]
+        TextMeshProUGUI m_TitleText;
+        [SerializeField]
+        TextMeshProUGUI m_MainText;
+
+        static PopupPanel s_Instance;
+
+        void Awake()
+        {
+            s_Instance = this;
```

Comment:
if you want a singleton, you might want to check if s_Instance is null here.

Reply:
Yes, but why use singleton when we have DI?

Reply:
Is DI the best use for this here? Every single class that'll want to show a popup will need to have an "Inject" method. while here, those classes just call popup.show. What would be the pros of using DI?
cc @LPLafontaineB

Reply:
Meh, I see you already did the change, why not. I would just be careful about blindly using DI everywhere. Let's go with what makes sense.

Reply:
Yes I just saw your comment, I can still revert it if we find using a singleton is better. Thinking back on it, since this might be used in a lot of different places, we might prefer to use a singleton instead, to avoid having to inject it in a lot of places when it will be used for a single method call.

Reply:
DI could be better than singletons in cases we want to unit test something. But here, this isn't really a true singleton, you could make sure you can destroy and recreate that monobehaviour and it'd work fine.
I could see a use for DI if we want to have different popups between main menu et boss room scene? Where we'd have different instances of popup, but only one at a time? But that seems like pre-optimization. If we reach that point, there's nothing preventing us from mixing singleton and DI too, where singleton would just return the injected instance.

Reply:
Wouldn't we want on the contrary to have popups persist from one scene to the other?

Reply:
We could have a different instance, with different visuals? That's a lot of "could" and "maybe", current situation is we have one single popup across multiple scenes.

Reply:
in that case, should this be reverted back? Since both work similarly, which version would be easier and cleaner for us to use, and for users?